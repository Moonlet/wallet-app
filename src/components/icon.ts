import { createIconSet } from 'react-native-vector-icons';
import * as glyphMap from '../../resources/fonts/icons.json';

Object.keys(glyphMap).map(key => {
    (glyphMap as any)[key] = parseInt(String((glyphMap as any)[key]).replace('\\f', '0xf'), 16);
});

export const Icon = createIconSet(glyphMap, 'icons', 'icons.ttf');
export default Icon;
