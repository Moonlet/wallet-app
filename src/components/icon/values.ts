export enum IconValues {
    // please try to add keys in alphabetic order
    ADD = 'add',
    ADD_CIRCLE = 'add-circle',
    ALARM_BELL = 'alarm-bell',
    ARCHIVE_LOCKER = 'archive-locker',
    ARROW_BUTTON_DOWN = 'arrow-button-down',
    ARROW_DOWN = 'arrow-down',
    ARROW_LEFT = 'arrow-left',
    ARROW_RIGHT_DASH = 'arrow-right-dash',
    ARROW_RIGHT = 'arrow-right',
    BIN = 'bin',
    BLE = 'ble',
    BUG = 'bug',
    BUTTON_REFRESH_ONE = 'button-refresh-one',
    CHECK_BOX_THICKED = 'check-box-thicked',
    CHECK_BOX = 'check-box',
    CHECK = 'check',
    CHEVRON_DOWN = 'chevron-down',
    CHEVRON_DOWN_BIGGER = 'chevron-down-2',
    CHEVRON_RIGHT = 'chevron-right',
    CHEVRON_UP = 'chevron-up',
    CLAIM_REWARD = 'claim-reward',
    CLOSE = 'close',
    COPY = 'copy',
    DASHBOARD = 'dashboard',
    EXPAND = 'expand',
    EYE = 'eye',
    FACE_ID = 'face-id',
    FAILED = 'failed',
    FLASH_OFF = 'flash-off',
    FLASH = 'flash',
    INBOUND = 'inbound',
    KEY_LOCK = 'key-lock',
    KEY = 'key',
    KEYBOARD_DELETE = 'keyboard-delete',
    KEYBOARD_SHIFT = 'keyboard-shift',
    LAPTOP = 'laptop',
    LEDGER_LOOGO = 'ledger-logo',
    MONEY_WALLET = 'money-wallet',
    MONITOR = 'monitor',
    MINUS = 'minus',
    NAVIGATION_MENU_HORIZONTAL = 'navigation-menu-horizontal',
    NAVIGATION_MENU_VERTICAL = 'navigation-menu-vertical',
    NAVIGATION_MENU = 'navigation-menu',
    NOTES_LIST = 'notes-list',
    OUTBOUND = 'outbound',
    PENCIL = 'pencil',
    PENDING = 'pending',
    PLUS = 'plus',
    QR_CODE_SCAN = 'qr-code-scan',
    REINVEST = 'reinvest',
    REVOTE = 'revote',
    SATURN_ICON = 'saturn-icon',
    SEARCH = 'search',
    SETTINGS = 'settings',
    SINGLE_MAN_HIERACHY = 'single-man-hierachy',
    TOUCH_ID = 'touch-id',
    UNLOCK = 'unlock',
    UNVOTE = 'unvote',
    USB = 'usb',
    VIEW_OFF = 'view-off',
    VIEW = 'view',
    VOTE = 'vote',
    WARNING = 'warning',
    WITHDRAW = 'withdraw'
    // please try to add keys in alphabetic order
}
