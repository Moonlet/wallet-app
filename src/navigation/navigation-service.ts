import { NavigationActions, StackActions, NavigationParams } from 'react-navigation';

export const NavigationService = (() => {
    let navigator;

    const setTopLevelNavigator = navigatorRef => {
        navigator = navigatorRef;
    };

    const navigate = (routeName: string, params: NavigationParams) => {
        navigator?.dispatch(
            NavigationActions.navigate({
                routeName,
                params
            })
        );
    };

    const replace = (routeName: string, params: NavigationParams) => {
        navigator?.dispatch(
            StackActions.replace({
                routeName,
                params
            })
        );
    };

    const popToTop = () => {
        navigator?.dispatch(StackActions.popToTop());
    };

    const goBack = () => {
        navigator?.dispatch(NavigationActions.back());
    };

    const getRecursiveRoute = routeState => {
        if (Array.isArray(routeState.routes)) {
            return getRecursiveRoute(routeState.routes[routeState.index]);
        } else {
            return routeState.routeName;
        }
    };

    const getRecursiveRouteWithParams = (routeState: any) => {
        if (Array.isArray(routeState.routes)) {
            return getRecursiveRouteWithParams(routeState.routes[routeState.index]);
        } else {
            return {
                routeName: routeState.routeName,
                params: routeState?.params
            };
        }
    };

    const getCurrentRoute = () => navigator && getRecursiveRoute(navigator.state.nav);

    const getCurrentRouteWithParams = () =>
        navigator && getRecursiveRouteWithParams(navigator.state.nav);

    return {
        setTopLevelNavigator,
        navigate,
        replace,
        popToTop,
        getCurrentRoute,
        goBack,
        getCurrentRouteWithParams
    };
})();
