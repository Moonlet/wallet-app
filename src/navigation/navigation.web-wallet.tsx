import React from 'react';
import { ConnectWebWalletScreen } from '../screens/connect-web/connect-web-wallet';
import { createStackNavigator } from '@react-navigation/stack';
import { DashboardScreen } from '../screens/dashboard/dashboard.web-wallet';
import { ChartDataWebScreen } from '../screens/chart-data-web/chart-data-web';
import { NavigationContainer } from '@react-navigation/native';
import { ITheme } from '../core/theme/itheme';
import { darkTheme } from '../styles/themes/dark-theme';
import { lightTheme } from '../styles/themes/light-theme';
import { Theme } from '../core/theme/themes';

export const themes: { dark: ITheme; light: ITheme } = {
    [Theme.dark]: darkTheme,
    [Theme.light]: lightTheme
};

const defaultNavigationOptions = () => ({
    gestureEnabled: false,
    headerShown: false
});

const Stack = createStackNavigator();
// Web Wallet navigation stack
export const WebWalletNavigation = ({ theme }) => (
    <NavigationContainer theme={theme}>
        <Stack.Navigator initialRouteName={'ConnectWebWalletScreen'}>
            <Stack.Screen
                options={defaultNavigationOptions}
                component={DashboardScreen}
                name={'DashboardScreen'}
            />
            <Stack.Screen
                options={defaultNavigationOptions}
                component={ConnectWebWalletScreen}
                name={'ConnectWebWalletScreen'}
            />
            <Stack.Screen
                options={defaultNavigationOptions}
                component={ChartDataWebScreen}
                name={'ChartDataWebScreen'}
            />
        </Stack.Navigator>
    </NavigationContainer>
);
