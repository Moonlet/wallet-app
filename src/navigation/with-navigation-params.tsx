import React from 'react';
import { NavigationScreenProp, NavigationState } from 'react-navigation';
import hoistNonReactStatics from 'hoist-non-react-statics';

export type INavigationProps<P = {}> = P & {
    navigation: NavigationScreenProp<NavigationState, P>;
};

export const withNavigationParams = (options?: { forwardRef?: boolean }) => (Comp: any) => {
    const displayName = `withNavigation(${Comp.displayName || Comp.name || 'Component'})`;
    function Component(props: any) {
        let params;
        if (props.navigation && props.navigation.state && props.navigation.state.params) {
            params = props.navigation.state.params;
        }
        return <Comp {...props} {...params} ref={props.forwardedRef} />;
    }
    Component.displayName = displayName;
    hoistNonReactStatics(Component, Comp);

    function ForwardRef(props, ref) {
        return <Component {...props} forwardedRef={ref} />;
    }

    if (options?.forwardRef) {
        const forwardedRef = React.forwardRef(ForwardRef);

        ForwardRef.displayName = displayName;
        hoistNonReactStatics(forwardedRef, Comp);
        return forwardedRef;
    }
    return Component;
};
