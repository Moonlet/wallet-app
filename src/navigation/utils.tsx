import React from 'react';
import { View, Text } from 'react-native';
import { Icon } from '../components/icon';
import { COLORS } from '../styles/colors';
import { StackViewStyleInterpolator } from 'react-navigation-stack';
import { normalize } from '../library';

export const DummyScreen = () => (
    <View style={{ display: 'flex', justifyContent: 'center', flex: 1 }}>
        <Text style={{ textAlign: 'center' }}>some screen</Text>
    </View>
);

// TODO use theme for colors
export const menuIcon = (icon: string) => ({ focused }: any) => (
    <Icon
        name={icon}
        size={normalize(25)}
        style={{ color: focused ? COLORS.AQUA : COLORS.DARK_GRAY }}
    />
);

// remove animation when transitioning to any of `noAnimationScreens`
export const removeAnimation = (noAnimationScreens: string[]) => sceneProps => {
    if (noAnimationScreens.indexOf(sceneProps.scene.route.routeName) !== -1) {
        return null;
    }
    return StackViewStyleInterpolator.forHorizontal(sceneProps);
};
