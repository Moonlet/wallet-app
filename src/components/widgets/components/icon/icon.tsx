import React from 'react';
import { TouchableOpacity } from 'react-native';
import stylesProvider from './styles';
import { smartConnect } from '../../../../core/utils/smart-connect';
import { withTheme, IThemeProps } from '../../../../core/theme/with-theme';
import { IScreenModule, IIconData, ISmartScreenActions } from '../../types';
import { formatStyles } from '../../utils';
import Icon from '../../../icon/icon';
import { normalize } from '../../../../styles/dimensions';

interface IExternalProps {
    module: IScreenModule;
    actions: ISmartScreenActions;
    options?: {
        screenKey?: string;
        flowId?: string;
    };
}

const IconModuleComponent = (
    props: IThemeProps<ReturnType<typeof stylesProvider>> & IExternalProps
) => {
    const { module, styles } = props;
    const data = module.data as IIconData;

    const customStyle = formatStyles(module?.style);

    const moduleJSX = (
        <Icon
            name={data.icon}
            size={(customStyle?.size as number) || normalize(18)}
            style={[styles.icon, customStyle]}
        />
    );

    if (module?.cta) {
        return (
            <TouchableOpacity
                onPress={() => props.actions.handleCta(module.cta, props?.options)}
                activeOpacity={0.9}
            >
                {moduleJSX}
            </TouchableOpacity>
        );
    } else {
        return moduleJSX;
    }
};

export const IconModule = smartConnect<IExternalProps>(IconModuleComponent, [
    withTheme(stylesProvider)
]);
