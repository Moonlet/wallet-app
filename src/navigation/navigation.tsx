import React from 'react';

import { createBottomTabNavigator } from 'react-navigation-tabs';
import { createStackNavigator, StackViewTransitionConfigs } from 'react-navigation-stack';
import { createSwitchNavigator } from 'react-navigation';

import { darkTheme } from '../styles/themes/dark-theme';
import { lightTheme } from '../styles/themes/light-theme';
import { ITheme } from '../core/theme/itheme';
import { Theme } from '../core/theme/themes';
import { HeaderLeft } from '../components/header-left/header-left';
import { menuIcon } from './utils';

import { DashboardScreen } from '../screens/dashboard/dashboard';
import { SettingsScreen } from '../screens/settings/settings';
import { NetworkOptionsScreen } from '../screens/settings/network-options/network-options';
import { NetworkSelectionScreen } from '../screens/settings/network-selection/network-selection';
import { SetCurrencyScreen } from '../screens/settings/set-currency/set-currency';
import { BlockchainPortfolioScreen } from '../screens/settings/blockchain-portfolio/blockchain-portfolio';
import { BackupWalletScreen } from '../screens/settings/backup-wallet/backup-wallet';
import { OnboardingScreen } from '../screens/onboarding/onboarding';
import { CreateWalletMnemonicScreen } from '../screens/create-wallet-mnemonic/create-wallet-mnemonic';
import { TokenScreen } from '../screens/token/token';
import { SendScreen } from '../screens/send/send';
import { TermsConditionsScreen } from '../screens/terms-conditions/terms-conditions';
import { PrivacyPolicyScreen } from '../screens/privacy-policy/privacy-policy';
import { CreateWalletConfirmMnemonicScreen } from '../screens/create-wallet-confirm-mnemonic/create-wallet-confirm-mnemonic';
import { RecoverWalletScreen } from '../screens/recover-wallet/recover-wallet';
import { WalletsScreen } from '../screens/wallets/wallets';
import { ReceiveScreen } from '../screens/receive/receive';
import { ViewWalletMnemonicScreen } from '../screens/view-wallet-mnemonic/view-wallet-mnemonic';
import { TransactionsHistoryScreen } from '../screens/transactions-history/transactions-history';
import { TransactionDetails } from '../screens/transaction-details/transaction-details';
import { StatisticsScreen } from '../screens/statistics/statistics';
import { WatchScreen } from '../screens/watch/watch';
import { ManageAccountScreen } from '../screens/manage-account/manage-account';
import {
    BASE_DIMENSION,
    normalize,
    normalizeFontAndLineHeight,
    LETTER_SPACING,
    normalizeLetterSpacing
} from '../styles/dimensions';
import { ConnectHardwareWallet } from '../screens/connect-hardware-wallet/connect-hardware-wallet';
import { ManageTokenScreen } from '../screens/manage-token/manage-token';
import { ConnectExtensionScreen } from '../screens/connect-extension/connect-extension';
import { ValidatorScreen } from '../screens/validator-screen/validator-screen';

interface IDefaultNavOptions {
    navigation: any;
    theme: Theme;
}

export const themes: { dark: ITheme; light: ITheme } = {
    [Theme.dark]: darkTheme,
    [Theme.light]: lightTheme
};

const mainTabbedNavigationOptions: any = {
    tabBarPosition: 'bottom',
    tabBarOptions: {
        showIcon: true,
        labelStyle: {
            fontSize: normalizeFontAndLineHeight(12),
            lineHeight: normalizeFontAndLineHeight(15),
            fontWeight: '500',
            letterSpacing: normalizeLetterSpacing(0.3)
        },
        style: {
            padding: BASE_DIMENSION,
            height: normalize(60)
        },
        indicatorStyle: {
            display: 'none'
        },
        activeTintColor: {
            dark: darkTheme.colors.accent
        },
        inactiveTintColor: {
            dark: darkTheme.colors.textSecondary
        }
    },

    swipeEnabled: false
};

export const defaultStackNavigationOptions: any = ({ navigation, theme }: IDefaultNavOptions) => ({
    headerStyle: {
        backgroundColor: themes[theme].colors.appBackground,
        elevation: 0,
        borderBottomWidth: 0
    },
    headerTitleStyle: {
        flex: 1,
        fontSize: normalize(22),
        lineHeight: normalize(28),
        color: themes[theme].colors.text,
        letterSpacing: LETTER_SPACING,
        textAlign: 'center',
        fontWeight: 'bold'
    },
    headerLeft: navigation.dangerouslyGetParent().state.index > 0 && (
        <HeaderLeft icon="arrow-left-1" onPress={() => navigation.goBack(null)} />
    )
});

// wallet navigation stack
export const WalletNavigation = createStackNavigator(
    {
        Dashboard: {
            screen: DashboardScreen
        },
        Token: {
            screen: TokenScreen
        },
        Send: {
            screen: SendScreen
        },
        Receive: {
            screen: ReceiveScreen
        },
        Wallets: {
            screen: WalletsScreen
        },
        ViewWalletMnemonic: {
            screen: ViewWalletMnemonicScreen
        },
        TransactonsHistory: {
            screen: TransactionsHistoryScreen
        },
        TransactionDetails: {
            screen: TransactionDetails
        },
        ManageAccount: {
            screen: ManageAccountScreen
        },
        ManageToken: {
            screen: ManageTokenScreen
        },
        ConnectExtension: {
            screen: ConnectExtensionScreen
        },
        Validator: {
            screen: ValidatorScreen
        },

        // wallet creation
        CreateWalletMnemonic: {
            screen: CreateWalletMnemonicScreen
        },
        TermsConditions: {
            screen: TermsConditionsScreen
        },
        PrivacyPolicy: {
            screen: PrivacyPolicyScreen
        },
        CreateWalletConfirmMnemonic: {
            screen: CreateWalletConfirmMnemonicScreen
        },
        RecoverWallet: {
            screen: RecoverWalletScreen
        },
        ConnectHardwareWallet: {
            screen: ConnectHardwareWallet
        }
    },
    {
        initialRouteName: 'Dashboard',
        defaultNavigationOptions: defaultStackNavigationOptions,
        headerLayoutPreset: 'center',
        navigationOptions: ({ navigation }) => ({
            tabBarVisible: navigation.state.index < 1
        }),
        transitionConfig: () => StackViewTransitionConfigs.SlideFromRightIOS
    }
);

// settings navigation stack
export const SettingsNavigation = createStackNavigator(
    {
        Settings: {
            screen: SettingsScreen
        },
        NetworkOptions: {
            screen: NetworkOptionsScreen
        },
        NetworkSelection: {
            screen: NetworkSelectionScreen
        },
        TermsConditions: {
            screen: TermsConditionsScreen
        },
        PrivacyPolicy: {
            screen: PrivacyPolicyScreen
        },
        Wallets: {
            screen: WalletsScreen
        },
        ViewWalletMnemonic: {
            screen: ViewWalletMnemonicScreen
        },
        SetCurrency: {
            screen: SetCurrencyScreen
        },
        BlockchainPortfolio: {
            screen: BlockchainPortfolioScreen
        },
        BackupWallet: {
            screen: BackupWalletScreen
        },

        // wallet creation
        CreateWalletMnemonic: {
            screen: CreateWalletMnemonicScreen
        },
        CreateWalletConfirmMnemonic: {
            screen: CreateWalletConfirmMnemonicScreen
        },
        RecoverWallet: {
            screen: RecoverWalletScreen
        },
        ConnectHardwareWallet: {
            screen: ConnectHardwareWallet
        }
    },
    {
        initialRouteName: 'Settings',
        defaultNavigationOptions: defaultStackNavigationOptions,
        headerLayoutPreset: 'center',
        navigationOptions: ({ navigation }) => ({
            tabBarVisible: navigation.state.index < 1
        }),
        transitionConfig: () => StackViewTransitionConfigs.SlideFromRightIOS
    }
);

// rewards navigation stack
export const StatisticsNavigation = createStackNavigator(
    {
        Statistics: {
            screen: StatisticsScreen
        }
    },
    {
        initialRouteName: 'Statistics',
        defaultNavigationOptions: defaultStackNavigationOptions,
        headerLayoutPreset: 'center',
        transitionConfig: () => StackViewTransitionConfigs.SlideFromRightIOS
    }
);

// watch navigation stack
export const WatchNavigation = createStackNavigator(
    {
        Watch: {
            screen: WatchScreen
        }
    },
    {
        initialRouteName: 'Watch',
        defaultNavigationOptions: defaultStackNavigationOptions,
        headerLayoutPreset: 'center',
        transitionConfig: () => StackViewTransitionConfigs.SlideFromRightIOS
    }
);

// main dashboard navigation
export const navigationConfig = {
    Dashboard: {
        screen: WalletNavigation,
        navigationOptions: () => ({
            tabBarIcon: menuIcon('dashboard')
        })
    },
    Statistics: {
        screen: StatisticsNavigation,
        headerTransparent: true,
        navigationOptions: () => ({
            tabBarIcon: menuIcon('reinvest')
        })
    },
    Watch: {
        screen: WatchNavigation,
        headerTransparent: true,
        navigationOptions: () => ({
            tabBarIcon: menuIcon('view-1')
        })
    },
    Settings: {
        screen: SettingsNavigation,
        headerTransparent: true,
        navigationOptions: () => ({
            tabBarIcon: menuIcon('cog')
        })
    }
};

// wallet creation flow stack
export const OnboardingNavigation = createStackNavigator(
    {
        Onboarding: {
            screen: OnboardingScreen
        },
        CreateWalletMnemonic: {
            screen: CreateWalletMnemonicScreen
        },
        TermsConditions: {
            screen: TermsConditionsScreen
        },
        PrivacyPolicy: {
            screen: PrivacyPolicyScreen
        },
        CreateWalletConfirmMnemonic: {
            screen: CreateWalletConfirmMnemonicScreen
        },
        RecoverWallet: {
            screen: RecoverWalletScreen
        },
        ConnectHardwareWallet: {
            screen: ConnectHardwareWallet
        }
    },
    {
        initialRouteName: 'Onboarding',
        defaultNavigationOptions: defaultStackNavigationOptions,
        headerLayoutPreset: 'center',
        transitionConfig: () => StackViewTransitionConfigs.SlideFromRightIOS
    }
);

// main bottom bar navigation
export const MainNavigation = createBottomTabNavigator(
    navigationConfig,
    mainTabbedNavigationOptions
);

export const RootNavigation = createSwitchNavigator(
    {
        MainNavigation,
        OnboardingNavigation
    },
    {
        initialRouteName: 'MainNavigation'
    }
);
