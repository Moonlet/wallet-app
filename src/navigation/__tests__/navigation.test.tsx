import { navigationConfig, defaultStackNavigationOptions } from '../navigation';
import { menuIcon } from '../utils';
import { IconValues } from '../../components/icon/values';

jest.mock('../utils');

jest.mock('../../redux/config');

export default describe('Navigation', () => {
    describe('configuration', () => {
        test('config is ok', () => {
            navigationConfig.Dashboard.navigationOptions();
            expect(menuIcon).toHaveBeenCalledWith(IconValues.DASHBOARD);
            (menuIcon as any).mockClear();

            navigationConfig.Watch.navigationOptions();
            expect(menuIcon).toHaveBeenCalledWith(IconValues.VIEW);
            (menuIcon as any).mockClear();

            navigationConfig.Statistics.navigationOptions();
            expect(menuIcon).toHaveBeenCalledWith(IconValues.REINVEST);
            (menuIcon as any).mockClear();

            navigationConfig.Settings.navigationOptions();
            expect(menuIcon).toHaveBeenCalledWith(IconValues.SETTINGS);
            (menuIcon as any).mockClear();
        });

        test('default stack configuration is ok', () => {
            const navigation = {
                dangerouslyGetParent: () => ({
                    state: {
                        index: 1
                    }
                }),
                goBack: jest.fn()
            };
            const defaultNavOptions = defaultStackNavigationOptions({ navigation, theme: 'dark' });

            defaultNavOptions.headerLeft.props.onPress();
            expect(navigation.goBack).toBeCalledWith(null);
            expect(defaultNavOptions).toMatchSnapshot();
        });

        test('default stack configuration should not display back button if there is nowhere to go back to', () => {
            const navigation = {
                dangerouslyGetParent: () => ({
                    state: {
                        index: 0
                    }
                })
            };
            const defaultNavOptions = defaultStackNavigationOptions({ navigation, theme: 'dark' });
            expect(defaultNavOptions).toMatchSnapshot();
        });
    });
});
