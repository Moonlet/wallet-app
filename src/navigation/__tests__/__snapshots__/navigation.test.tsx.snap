// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation configuration default stack configuration is ok 1`] = `
Object {
  "headerLeft": <withTheme(HeaderLeftComponent)
    icon="arrow-left-1"
    onPress={[Function]}
  />,
  "headerStyle": Object {
    "backgroundColor": "#121212",
    "borderBottomWidth": 0,
  },
  "headerTitleStyle": Object {
    "color": "#E0E0E0",
    "flex": 1,
    "fontSize": 20,
    "letterSpacing": 0.38,
    "lineHeight": 25,
    "textAlign": "center",
  },
}
`;

exports[`Navigation configuration default stack configuration should not display back button if there is nowhere to go back to 1`] = `
Object {
  "headerLeft": false,
  "headerStyle": Object {
    "backgroundColor": "#121212",
    "borderBottomWidth": 0,
  },
  "headerTitleStyle": Object {
    "color": "#E0E0E0",
    "flex": 1,
    "fontSize": 20,
    "letterSpacing": 0.38,
    "lineHeight": 25,
    "textAlign": "center",
  },
}
`;
